tests/components/test_nlp.py

```diff
@@ -316,6 +316,67 @@ def test_allowed_entities(kwik_e_mart_nlp):
     assert res['entities'][0]['text'] == 'xyz'
 
 
+def test_disallowed_entities(kwik_e_mart_nlp):
+    res = kwik_e_mart_nlp.process("hello")
+    assert res['intent'] == 'greet'
+    res = kwik_e_mart_nlp.process("hello", deny_nlp=["store_info.greet"])
+    assert res['intent'] != 'greet'
+
+    res = kwik_e_mart_nlp.process("transfer $200 from checking to savings")
+    assert res['intent'] == 'transfer_money'
+    assert res['entities'][0]['type'] == 'sys_amount-of-money'
+    assert res['entities'][1]['type'] == 'account_type'
+    assert res['entities'][1]['role'] == 'origin'
+    assert res['entities'][2]['type'] == 'account_type'
+    assert res['entities'][2]['role'] == 'dest'
+
+    res = kwik_e_mart_nlp.process("transfer $200 from checking to savings",
+                                  deny_nlp=["banking.transfer_money"])
+    assert res['intent'] != 'transfer_money'
+
+    res = kwik_e_mart_nlp.process("transfer $200 from checking to savings",
+                                  deny_nlp=["banking.transfer_money.account_type"])
+    assert res['intent'] == 'transfer_money'
+    assert 'account_type' not in {entity['type'] for entity in res['entities']}
+
+    res = kwik_e_mart_nlp.process("please can you tell me if springfield is "
+                                  "possibly open at this time on friday",
+                                  allow_nlp=["store_info"])
+    assert res['intent'] == 'get_store_hours'
+    assert 'sys_time' in {entity['type'] for entity in res['entities']}
+
+    res = kwik_e_mart_nlp.process("please can you tell me if springfield is "
+                                  "possibly open at this time on friday",
+                                  allow_nlp=["store_info"],
+                                  deny_nlp=["store_info.get_store_hours.sys_time"])
+    assert res['intent'] == 'get_store_hours'
+    assert 'sys_time' not in {entity['type'] for entity in res['entities']}
+
+    # If all entities in an intent are denied, that intent is also denied.
+    res = kwik_e_mart_nlp.process("please can you tell me if springfield is "
+                                  "possibly open at this time on friday",
+                                  allow_nlp=["store_info"],
+                                  deny_nlp=["store_info.get_store_hours.sys_time",
+                                            "store_info.get_store_hours.store_name"])
+    assert res['intent'] != 'get_store_hours'
+    assert 'sys_time' not in {entity['type'] for entity in res['entities']}
+    assert 'store_name' not in {entity['type'] for entity in res['entities']}
+
+    res = kwik_e_mart_nlp.process("please can you tell me if springfield is "
```

Comment:
Should we also test the case where allow_nlp is a subset of deny_nlp? For Example: `allow_nlp = store_info.get_store_hours` but `deny_nlp = store_info`

Reply:
done